File: apollo-normalized-cache/src/main/java/com/apollographql/apollo/internal/cache/normalized/ReadableStore.java
package com.apollographql.apollo.internal.cache.normalized;

import com.apollographql.apollo.cache.CacheHeaders;
import com.apollographql.apollo.cache.normalized.Record;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.Collection;

public interface ReadableStore {

  @Nullable Record read(@NotNull String key, @NotNull CacheHeaders cacheHeaders);

  Collection<Record> read(@NotNull Collection<String> keys, @NotNull CacheHeaders cacheHeaders);

}

File: apollo-normalized-cache-api/src/commonMain/kotlin/com/apollographql/apollo3/cache/normalized/internal/ReadableStore.kt
package com.apollographql.apollo3.cache.normalized.internal

import com.apollographql.apollo3.cache.CacheHeaders
import com.apollographql.apollo3.cache.normalized.Record

interface ReadableStore {
  fun read(key: String, cacheHeaders: CacheHeaders): Record?
  fun read(keys: Collection<String>, cacheHeaders: CacheHeaders): Collection<Record>
}
